Feature/login_na

nahokyun opened the pull request:
## Login관련 기능 처리

1. 로그인관련 요청을 처리하는 LoginController 클래스를 생성하였습니다.

2. MemberMapper에는 email인증여부를 체크하는 checkEmailAuth, 로그인시 멤버의 정보를 가져오는 loadUserInfo와 암호화되어있는 비밀번호를 가져오는 findPasswordByLoginId가 추가되었습니다.

3. Session 정보의 Id가될 SessionKey 클래스를 생성하였습니다.

nahokyun committed:
feature/login_na

로그인관련 요청을 처리하는 LoginController 클래스를 생성하였습니다.

MemberMapper에는 email인증여부를 체크하는 checkEmailAuth, 로그인시 멤버의 정보를 가져오는 loadUserInfo와 암호화되어있는 비밀번호를 가져오는 findPasswordByLoginId가 추가되었습니다.

Session 정보의 Id가될 SessionKey 클래스를 생성하였습니다.

sunwootest commented on src/main/java/study/movieservice/controller/LoginController.java:
```diff
@@ -0,0 +1,43 @@
+
+    @PostMapping("/login")
+    public ResponseEntity<String> loginAccess(@RequestBody Member member){
+        Optional<Member> receiver= memberService.findByLoginId(member.getLoginId());
+
+        if(receiver.isPresent()){
+            if(memberService.checkEmailAuth(member.getLoginId())!=1) {
+                return new ResponseEntity<>(NOT_COMPLETED_AUTH.getText(),NOT_COMPLETED_AUTH.getStatus());
+            }
+
+            if(memberService.loginAccess(member)) {
+                return new ResponseEntity<>(SUCCESS_LOGIN.getText(), SUCCESS_LOGIN.getStatus());
+            }
+
+            return new ResponseEntity<>(FAILED_LOGIN.getText(),FAILED_LOGIN.getStatus());
+        }
+        else
+            return new ResponseEntity<>(CANNOT_FOUND_ID.getText(), CANNOT_FOUND_ID.getStatus());
```
presentation layer에 담길 내용들은 아닌 것 같습니다.

sunwootest replied:
실패관련 로직은 Exception Handler로 처리하는 것도 괜찮아보이네요

nahokyun replied:
적절히 있어야할 곳으로 로직을 분리하고 Exception Handler를 보완해서 적용해보겠습니다.

sunwootest commented on src/main/java/study/movieservice/domain/SessionKey.java:
```diff
@@ -0,0 +1,8 @@
+
+import lombok.Getter;
+
+@Getter
+public class SessionKey {
+    public static final String SESSION_LOGIN_ID="loginId";
+}
```
Getter가 필요한가요?

nahokyun replied:
굳이 필요하지않을것 같습니다. 수정하겠습니다.

sunwootest commented on src/main/java/study/movieservice/service/MemberService.java:
```diff
@@ -44,4 +47,27 @@ public boolean updateMailAuth(Member member, String email, String emailKey) {
         return false;
     }
+
+    public int checkEmailAuth(String loginId){
+        return memberMapper.checkEmailAuth(loginId);
+    }
```
해당 메소드를 int로 반환하는 것보다 더 나은 대안이 있을 것 같습니다.

nahokyun replied:
boolean으로 반환하는 방법을 좀더 알아보고 수정하도록 하겠습니다.

sunwootest commented on src/main/java/study/movieservice/service/MemberService.java:
```diff
@@ -4,15 +4,18 @@
 public class MemberService {
 
     private final MemberMapper memberMapper;
+    private final HttpSession httpSession;
```
MemberService가 세션에 의존하게 되었네요.

nahokyun replied:
LoginService를 만들어 세션을 필요한곳에서만 사용할 수 있도록 수정해보겠습니다.

sunwootest reviewed (commented):
기본적인 로직은 잘 짜셨습니다. 코드만 잘 다듬으면 더 나아질 것 같네요 :)

nahokyun committed:
feature/login_na

presentation layer에 적절하지않은 내용들을 LoginService를 추가하여 분리하였습니다.

ExceptionHandler를 보완하여 인증이 진행되지않았을때와 찾는멤버가 없을때를 throw하여 처리하였습니다.

checkEmailAuth의 반환형을 boolean로 변경하고 MemberMapper에 접근하는 메소드들을 MemberService에 추가하여 LoginService에서 MemberService를 통해서만 데이터에 접근할수있게 구현하였습니다.

nahokyun committed:
feature/login_na

불필요한 주석 삭제

nahokyun commented:
### 추가된 내용
1. custom exception을 추가하여 인증여부와 멤버존재여부를 관리하였습니다.
### 변경된 내용
1. contoller영역에 불필요한 로직을 service영역으로 이동시켰습니다. 
### 삭제된 내용
1. 불필요한 주석, 어노테이션 삭제

sunwootest commented:
라이브 코드리뷰 했던 내용들 반영해주세요

nahokyun committed:
feature/login_na

범용적인 사용을 위해 클래스명을 변경 하였습니다.

쿼리 내에서 메일 인증여부를 비교하는것이 아니라 값을 가져온뒤 자바 코드내에서 비교하는것으로 로직을 변경하였습니다.

nahokyun commented:
> ### 변경된점
1. 한정적인 의미였던 NotCompleteAuthException 과 NotFoundMemberException을 rename 하여 NotCompleteException과 NotFoundException으로 변경하였습니다.

2. MemberMapper.xml과 MemberService 수정 (쿼리 내에서 비교하는것이 아닌 email_auth값을 직접 받아와서 자바코드 내에서 비교하는것으로 변경하였습니다.)
##  MemberMapper.xml
### 변경 전
```
   <select id="checkEmailAuth" resultType="boolean">
        select if(count(*)=1,1,0)
        from member
        where login_id = #{loginId} and email_auth=1
    </select>
```
### 변경 후 
```
    <select id="checkEmailAuth" resultType="int">
        select email_auth
        from member
        where login_id =#{loginId}
    </select>
```
## MemberService.java
### 변경 전 
```
    public boolean checkEmailAuth(String loginId){
        return memberMapper.checkEmailAuth(loginId);
    }
```
### 변경 후
```
    public boolean checkEmailAuth(String loginId){
        int compareEmailAuth = memberMapper.checkEmailAuth(loginId);
        if(compareEmailAuth==0)
            return false;
        return true;
    }
```
